src/gc-object-layout.h

```diff
@@ -0,0 +1,199 @@
+// This file is a part of Julia. License is MIT: https://julialang.org/license
+
+#ifndef JL_GC_OBJECT_LAYOUT_H
+#define JL_GC_OBJECT_LAYOUT_H
```

Comment:
I don't see the object layout types as part of the GC subsystem and I don't think this is the right name for this file.